pkg/provider-local/webhook/dnsconfig/mutator.go

```diff
@@ -38,30 +39,53 @@ func (m *mutator) InjectClient(c client.Client) error {
 }
 
 func (m *mutator) Mutate(ctx context.Context, newObj, oldObj client.Object) error {
-	if oldObj != nil {
-		// This is basically a hack - ideally, we would like the mutating webhook configuration to only react for CREATE
-		// operations. However, currently both "CREATE" and "UPDATE" are hard-coded in the extensions library.
-		return nil
-	}
+	var (
+		podMeta *metav1.ObjectMeta
+		podSpec *corev1.PodSpec
+	)
+
+	switch obj := newObj.(type) {
+	case *corev1.Pod:
+		if oldObj != nil {
+			// This is basically a hack - ideally, we would like the mutating webhook configuration to only react for CREATE
+			// operations. However, currently both "CREATE" and "UPDATE" are hard-coded in the extensions library.
+			return nil
+		}
+
+		if newObj.GetLabels()["app"] == "dependency-watchdog-probe" {
+			// We don't want to react for DWD pods but only for DWD deployments, so exit early here.
+			return nil
+		}
+
+		podMeta = &obj.ObjectMeta
+		podSpec = &obj.Spec
+
+	case *appsv1.Deployment:
+		podMeta = &obj.Spec.Template.ObjectMeta
+		podSpec = &obj.Spec.Template.Spec
 
-	pod, ok := newObj.(*corev1.Pod)
-	if !ok {
-		return fmt.Errorf("unexpected object, got %T wanted *corev1.Pod", newObj)
+	default:
+		return fmt.Errorf("unexpected object, got %T wanted *appsv1.Deployment or *corev1.Pod", newObj)
 	}
 
 	service := &corev1.Service{}
 	if err := m.client.Get(ctx, client.ObjectKey{Namespace: "gardener-extension-provider-local-coredns", Name: "coredns"}, service); err != nil {
 		return err
 	}
 
-	metav1.SetMetaDataLabel(&pod.ObjectMeta, local.LabelNetworkPolicyToIstioIngressGateway, v1beta1constants.LabelNetworkPolicyAllowed)
-	pod.Spec.DNSPolicy = corev1.DNSNone
-	pod.Spec.DNSConfig = &corev1.PodDNSConfig{
+	metav1.SetMetaDataLabel(podMeta, local.LabelNetworkPolicyToIstioIngressGateway, v1beta1constants.LabelNetworkPolicyAllowed)
+	m.injectDNSConfig(podSpec, newObj.GetNamespace(), service.Spec.ClusterIP)
+	return nil
+}
+
+func (m *mutator) injectDNSConfig(podSpec *corev1.PodSpec, namespace, coreDNSClusterIP string) {
```

Comment:
Nit: receiver not used.
```suggestion
func injectDNSConfig(podSpec *corev1.PodSpec, namespace, coreDNSClusterIP string) {
```